File: 0053-maximum-subarray/0053-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int max_so_far = INT_MIN, max_here = 0;
        int n = nums.size();
        for(int i=0; i<n; i++)
        {
            max_here += nums[i];
            if(max_so_far < max_here)
                max_so_far = max_here;

            if(max_here < 0)
                max_here =0;
        }
        return max_so_far;
    }
};

File: 0053-maximum-subarray/0053-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {

        int n=nums.length;
        int currMax = 0;
        int totalSum = Integer.MIN_VALUE;

        for(int i=0; i<n; i++) {
            currMax += nums[i];
            totalSum = Math.max(totalSum, currMax);
            
            if(currMax<0) currMax=0;
        }
        return totalSum;
    }
}